lib/matplotlib/ticker.py

```diff
@@ -377,19 +378,34 @@ class FuncFormatter(Formatter):
     """
     Use a user-defined function for formatting.
 
-    The function should take in two inputs (a tick value ``x`` and a
-    position ``pos``), and return a string containing the corresponding
-    tick label.
+    The function can take in at most two inputs (a required tick value ``x`` and 
+    an optional position ``pos``), and must return a string containing the 
+    corresponding tick label.
     """
     def __init__(self, func):
-        self.func = func
+        try:
+            self.nargs = len(inspect.signature(func).parameters)
+            if self.nargs == 2:
+                self.func = func 
+            elif self.nargs == 1:
+                def func_pos(x, pos):
+                    return func(x)
+                self.func = func_pos
+            else:
+                raise TypeError("""FuncFormatter functions take at most 
+                                2 parameters: x (required), pos (optional)""")
+        except ValueError as e:
+            #built ins like str.format don't have signatures
+            self.func = func
```

Comment:
I'd keep catching `ValueError` limited to parsing the signature.

```suggestion
        try:
            self.nargs = len(inspect.signature(func).parameters)
        except ValueError as e:
            #built ins like str.format don't have signatures
        else:
            self.func = func
            if self.nargs == 2:
                self.func = func 
            elif self.nargs == 1:
                def func_pos(x, pos):
                    return func(x)
                self.func = func_pos
            else:
                raise TypeError("""FuncFormatter functions take at most 
                                2 parameters: x (required), pos (optional)""")
```